### DISCUSSION TITLE
Node relations

### DISCUSSION THREAD

**Discussion by dhardy** [AccessKit/accesskit#284] [Q&A] [answered]
Hi! This question relates to integration with https://github.com/kas-gui/kas/

### Node relations

Nodes may be annotated with relations to other nodes: controls, details, "described by", "flow to", "labelled by", "radio group". How important are these things? Because a key part of the design of many modern toolkits, including Kas, is *avoiding* these to allow purely declarative widget trees.

The same goes for many other "relations" such as table header and next focus (focus stepping logic is outside the scope of widget definitions, though container widgets do play a role in it).

What are the consequences of not describing such relations?

Is <kbd>Tab</kbd> navigation normally provided only by the widget toolkit or by some interaction between that and the accessibility layer?

**Comment by mwcampbell** [comment, marked as answer]
Tab navigation is normally provided only by the toolkit.

I don't know yet if I'll keep all of these node relations; I mostly copied them from Chromium's internal accessibility definitions. The only one that AccessKit actually pays attention to so far is `labelled_by`.

**Comment by youssefx99** [comment]
yes of course
